### PR TITLE
Optimized filter operators; Added eslint; Added chunk operator; Altered README

### PR DESCRIPTION
**Pull request by jdpigeon**
This is a pretty big merge, with all the changes I've made this week to get the library working in my project. 

I've got all of the filter operators to a good working state based on our discussion https://github.com/alexcastillo/eeg-pipes/issues/2

Added eslint to make development nicer

I've started developing with the proposed data structures in mind. https://github.com/alexcastillo/eeg-pipes/issues/4. That includes replacing the groupByChannel operator with the chunk operator. I did my best to make all of those changes not impact anything that would have been previously written with the FFT operators. However, I never got around to testing those operators with these changes

### CODE REVIEW

**Comment by alexcastillo** [comment]
This is great, thank you! I'll be reviewing this in the next couple of days.

File: src/pipes/utility/chunk.js [hunk 1]
```diff
@@ -7,11 +7,11 @@ import {
   DATA_PROP as defaultDataProp,
   SAMPLE_RATE as defaultSamplingRate
 } from "../../constants";
+import { sample } from "rxjs/operators/sample";
```
Comment (alexcastillo) [inline src/pipes/utility/chunk.js]: Where is sample being used?

File: src/pipes/filtering/bandpassFilter.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    BW
+  };
+  const bandpassArray = new Array(nbChannels)
+    .fill(0)
+    .map(() => createBandpassIIR(options));
+  return createPipe(
+    source,
+    map(eegObject => {
+      // Should run multiStep function if inc. data is a Chunk (2D data array)
+      if (Array.isArray(eegObject.data[0])) {
```
Comment (alexcastillo) [inline src/pipes/filtering/bandpassFilter.js]: These two returns could be combined like:

```js 
const isChunk = Array.isArray(eegObject.data[0]);
const stepFunction = isChunk ? 'multiStep' : 'singleStep';
return {
  ...eegObject,
  data: eegObject.data.map((channel, 
    bandpassArray[index][stepFunction](channel)
  )
};
```

Comment (jdpigeon) [reply src/pipes/filtering/bandpassFilter.js]: Very nice. I didn't know you could do that with the strings as the function name

**Pull request merged**